test/keap/read/main.go

```diff
@@ -0,0 +1,42 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/amp-labs/connectors"
+	"github.com/amp-labs/connectors/common"
+	"github.com/amp-labs/connectors/providers/keap"
+	connTest "github.com/amp-labs/connectors/test/keap"
+	"github.com/amp-labs/connectors/test/utils"
+)
+
+func main() {
+	// Handle Ctrl-C gracefully.
+	ctx, done := signal.NotifyContext(context.Background(), syscall.SIGINT, syscall.SIGTERM)
+	defer done()
+
+	// Set up slog logging.
+	utils.SetupLogging()
+
+	conn := connTest.GetKeapConnector(ctx)
+	defer utils.Close(conn)
+
+	res, err := conn.Read(ctx, common.ReadParams{
+		ObjectName: "emails",
+		Fields:     connectors.Fields("id", "subject", "sent_from_address"),
+	})
+	if err != nil {
+		utils.Fail("error reading from microsoft CRM", "error", err)
+	}
+
+	fmt.Println("Reading emails..")
+	utils.DumpJSON(res, os.Stdout)
+
+	if res.Rows > keap.DefaultPageSize {
```

Comment:
Is this something you expect to happen?

Reply:
I keep up with the established purpose of the `test` directory which holds test. There is no good assert for reading and thats what we have for 7 deep connectors.
![image](https://github.com/user-attachments/assets/2e53be1a-9372-4464-90ce-0169b4743e9d)

Honestly, test scripts are rather Read/Write/ListObjectMetadata examples or just plain scripts using combination of those methods to demonstrate functionality.